### DISCUSSION TITLE
Subscenes and GameScene

### DISCUSSION THREAD

**Discussion by SergeMerzliakov** [AlmasB/FXGL#1105] [Q&A]
I have written a game demo where the game launches and uses subscenes for all the setup stuff - main screen, options,  and create game screen. Once I have created a game on the "create game" subscene (entities and components added to game world etc) I want to show the GameScene - but I cannot hide the subscenes.

here is my startup logic which goes straight to main subscene (with new game, resume game, options buttons)

		class Application : GameApplication() {
			...
			override fun initUI() {
				FXGL.run({
					mainController.showSubScene()
				}, Duration.seconds(0.0))
			}
			...
		}

How do I get subscenes and GameScene to interact properly? That is when playing the game showing GameScene, otherwise show subscenes?

**Update: I am looking at Breakout and other FXGL game samples for ideas**

**Comment by AlmasB** [comment]
> How do I get subscenes and GameScene to interact properly? That is when playing the game showing GameScene, otherwise show subscenes?

By default, the game enters GameScene, so there is no need to manually switch to it. Using `FXGL.getSceneService().***` one can push / pop scenes as required. I sense I'm missing the details here, so if this is not the answer you seek, please provide a bit more details on the use case.

**Reply by SergeMerzliakov** [reply]
You have it correct. My use case is I need a lot of subscenes for configuration and data screen (a text-heavy simulation type game - Football Manager is a good example). After looking at samples and wiki I figured out that GameScene is there on startup. I am using `FXGL.getSceneService().pop` until the current scene is the GameScene (checking with `FXGL.getSceneService().currentScene`). The only problem is that one of the subscenes is still showing on the screen. This is what threw me off and caused me to raise the issue here. It seems like I need to setup the GameView for the GameScene to get rid of subscene stuff. Will keep investigating.
